ea-2021-04-ndvi-automation-brian-pinke.py

```diff
@@ -0,0 +1,790 @@
+# * Create a sorted list of all tif bands
+# * Select the specific landsat scene desired
+# * Open the bands at the scene
+# * Open and clip the cloud mask layer using rasterio and clip
+# * Loop through bands for desired scene using open bands function
+# * Store bands in a list and calculate mean ndvi from mndvi function
+# * Use temp_list function to create list of desired attributes
+# * Create pandas df of landsat scene from list
+# 
+# ###### Task 2
+# * In this case we desire all scenes from all sites defined in the parent path
+# * Initial path steps are completed in prior task.
+# * Loop through all sites in the sites list (2 sites). For each loop:
+#     * Get site name 
+#     * Open shapefile of site for clipping using os and gpd
+#     * Get subdirectories of the site and create path to all scenes
+#     * Loop through all scenes at each site. For each loop:
+#         * Create sorted list of all bands in scene
+#         * Get the cloud mask layer for scene and clip with rasterio
+#         * Loop through the two bands per scene. For each loop:
+#             * Open and clean the two bands using open_clean_bands
+#             * Append opened bands to outer list: all_bands
+#         * Calculate mean ndvi of single scene using calc_mndvi
+#         * Create list of site name, scene date, scene mean ndvi 
+#         * Append this list to outermost list: all_list
+# * Use all_list, a list of lists containing site name, date and mean ndvi, to create a pandas dataframe
+# * Set dataframe date column to datetime format and set date to index
+# * Return completed ndvi_df dataframe.
+# 
+# ###### Task 3
+# * Create a new df without nan values
+# * Create plot
+# * Group df by site name and loop through:
+#     * plot x-axis as date
+#     * plot y-axis as mean ndvi
+#     * set label as site name
+# * Set axis labels and x-axis boundaries
+# * Set x-axis format to month and create plot grid
+# 
+# ###### Bonus Task
+# * Define desired directory location for .csv file
+# * Check if location exists
+#     * If it exists, continue to end of cell
+#     * If doesn't exist, make the directory location and continue to end of cell
+# * use .to_csv to create csv file at desired directory location
+#     
+
+# In[1]:
+
+
+# Autograding imports - do not modify this cell
+import matplotcheck.autograde as ag
+import matplotcheck.notebook as nb
+import matplotcheck.timeseries as ts
+from datetime import datetime
+
+
+# In[2]:
+
+
+# Import needed packages in PEP 8 order
+# and no unused imports listed (10 points total)
+
+# YOUR CODE HERE
+# Import necessary packages
+import os
+from glob import glob
+
+import matplotlib.pyplot as plt
+import numpy as np
+import geopandas as gpd
+import rioxarray as rxr
+import xarray as xr
+import earthpy as et
+import pandas as pd
+
+
+import matplotlib.dates as mdates
+from matplotlib.dates import DateFormatter
+
+
+# Get data and set working directory
+data = et.data.get_data('ndvi-automation')
```

Comment:
Good job on getting the data first before setting the working directory!

Reply:
I just learned a couple weeks ago that getting the data automatically creates the path...much more convenient than the if statement checking for sure.